core/src/main/java/nu/yona/server/messaging/service/MessageService.java

```diff
@@ -18,9 +20,9 @@
 import org.springframework.stereotype.Component;
 import org.springframework.stereotype.Service;
 
+import nu.yona.server.crypto.PublicKeyEncryptor;
```

Comment:
By concept, the ``MessageSource`` / ``MessageDestination`` pair is a store of encrypted messages.  That responsibility can well be implemented in these two entities. The logic is very limited..